### PR TITLE
Image hoist

### PR DESCRIPTION
**Pull request by stristr**
OK to merge.

Short review.

Summary of changes:

- Fixes [Large raster images in Sketch are virtually unusable](https://app.asana.com/0/785272717735380/704795620573335/f) and a host of related performance issues.
- Optimizes `mergeDesigns` by only calling `File.readMana` on slices that actually need to be merged.
- Changes the schema for locally hosted assets from `HAIKU_LOCAL_PROJECT_ROOT:` to `web+haikuroot://`, a proper schema that we can intercept for stuff like showing project thumbnails.

Regressions to look for:

- Problems with `mergeDesigns` downstream of the SVG replacement (but probably not an issue because we do all our FS stuff synchronously on the server side).
- Problems with existing projects containing image slices.
- Unanticipated problems from hoisting `xlink:href` from `attributes` (in mana) up to `timelines` (though none expected).

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [x] Wrote an automated test for existing and modified functionality
- [x] Updated `changelog/public/latest.json`

### CODE REVIEW

File: packages/@haiku/core/src/HaikuComponent.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ const KEYFRAME_ZERO = 0;
 const OBJECT = 'object';
 const MAX_INT = 2147483646;
 const SCOPE_STRATA = {div: 'div', svg: 'svg'};
-const CDN_ROOT_STR = 'HAIKU|CDN|PROJECT|ROOT'; // Split to avoid server-side munging
+const CDN_ROOT_STR = 'HAIKU|CDN|PROJECT|ROOT'.split('|').join('_'); // Split to avoid server-side munging
```
Comment (tsortwehttam) [inline packages/@haiku/core/src/HaikuComponent.ts:52]: Are you 100% sure this works after bundling? I believe that I did this initially, but webpack or rollup or minification or some bundling step was smart enough to identify the result as a constant and inlined the final (computed) string, breaking the functionality.

File: packages/haiku-glass/src/react/Preview.js [hunk 2]
```diff
@@ -106,12 +107,6 @@ export default class Preview extends React.Component {
   }
 }
 
-const ensureTrailingSlash = (str) => {
```
Comment (tsortwehttam) [inline packages/haiku-glass/src/react/Preview.js:109]: 🙏

File: packages/haiku-serialization/src/bll/Design.js [hunk 3]
```diff
@@ -1,62 +0,0 @@
-const path = require('path')
```
Comment (tsortwehttam) [inline packages/haiku-serialization/src/bll/Design.js:1]: 🙏 I've been wanting to do this for a long time.

**Comment by tsortwehttam** [comment]
LGTM -- with the provision to check against my note about that split string.

File: packages/@haiku/core/src/HaikuComponent.ts (hunk 1 above)
Comment (stristr) [reply packages/@haiku/core/src/HaikuComponent.ts:52]: With rollup (`index.standalone.js`), I didn't have any issues.

I also checked this against `create-react-app` via `share-page` without issue.

Comment (stristr) [reply packages/@haiku/core/src/HaikuComponent.ts:52]: (which uses webpack)

**Comment by stristr** [comment]
https://cl.ly/64245bde5da7 (index.standalone.js)
https://cl.ly/b49ef56a016e (bundle from `webpack`)

…so merging. Thanks for review.

**Pull request merged**